f3df/src/renderables.rs

```diff
@@ -192,17 +211,21 @@ pub fn convert_sector(sector: &crate::Sector) -> Vec<Face> {
                 } else {
                     Vector3::new(increment * count, increment, increment)
                 };
-                instance_data.push(Face {
+                faces.push(Face {
                     color: normalize_color(color),
-                    tree_index,
+                    tree_index: tree_index as f32,
                     normal,
                     matrix: compose_face_matrix(&center, &offset, &scale, &rotation),
                 });
             }
         }
     }
 
-    instance_data
+    Sector {
+        faces,
+        node_id_to_tree_index_map,
+        tree_index_to_node_id_map,
```

Comment:
Is publishing two maps the most efficient approach, or should we just have one array of (treeIndex -> nodeId)?

What guarantees do we have for `treeIndices` within a sector? Will they be continous?

Reply:
I do not know, but I was thinking that building the map in JS-land would be slower than doing it in Rust? If we only pass the `tree_index_to_node_id_map`, then we need to build the other one in JS somehow.

I do not know the guarantees we have for `treeIndices`, but I am pretty sure they are not continuous, because they depend on the tree structure from the original file, while sectors depend on the spatial structure of the geometry.

Reply:
Hmm, can RS Maps be transfered without serialization to JS? Or does the map need to be rebuilt in JS-land?

Reply:
We currently use serialization, but it should be possible to create the JS `js_sys::Map` in Rust and fill it directly. I am not sure how efficient either approach is, but it is just a bit more awkward to work with `js_sys::Map` than `std::collections::HashMap` in Rust. I was thinking to wait until we have some benchmarks before converting at least.